bot/exts/moderation/infraction/_utils.py

```diff
@@ -92,17 +92,25 @@ async def post_infraction(
 
     log.trace(f"Posting {infr_type} infraction for {user} to the API.")
 
+    current_time = datetime.now(tz=timezone.utc)
+
     payload = {
         "actor": ctx.author.id,  # Don't use ctx.message.author; antispam only patches ctx.author.
         "hidden": hidden,
         "reason": reason,
         "type": infr_type,
         "user": user.id,
         "active": active,
-        "dm_sent": dm_sent
+        "dm_sent": dm_sent,
+        "last_applied": current_time.isoformat(),
     }
```

Comment:
Just to make sure I understand, you're supplying `last_applied`, but what sets `inserted_at`? I thought the goal was to make the two identical in the beginning. Is it done on the site's end?

Reply:
Currently, the site is setting `inserted_at` when the entry is inserted. The goal of the new attribute `last_applied` is to provide a datetime that is calculated on command run at the bot side, which should remove some of the server / network latency issues. In addition, the `last_applied` time is actually calculated together with the delta, meaning that the absolute delta of `last_applied` and `expires_at` is exactly equal to the infraction (for example 1 hour) down to the microsecond.